spec/requests/api/v1/customers/usage_spec.rb

```diff
@@ -0,0 +1,270 @@
+      )
+
+      aggregate_failures do
+        expect(response).to have_http_status(:success)
+
+        expect(json[:customer_usage][:from_date]).to eq(Time.zone.today.beginning_of_month.iso8601)
+        expect(json[:customer_usage][:to_date]).to eq(Time.zone.today.end_of_month.iso8601)
+        expect(json[:customer_usage][:issuing_date]).to eq(Time.zone.today.end_of_month.iso8601)
+        expect(json[:customer_usage][:amount_cents]).to eq(5)
+        expect(json[:customer_usage][:amount_currency]).to eq('EUR')
+        expect(json[:customer_usage][:total_amount_cents]).to eq(6)
+        expect(json[:customer_usage][:total_amount_currency]).to eq('EUR')
+        expect(json[:customer_usage][:vat_amount_cents]).to eq(1)
+        expect(json[:customer_usage][:vat_amount_currency]).to eq('EUR')
+
+        charge_usage = json[:customer_usage][:charges_usage].first
+        expect(charge_usage[:billable_metric][:name]).to eq(metric.name)
+        expect(charge_usage[:billable_metric][:code]).to eq(metric.code)
+        expect(charge_usage[:billable_metric][:aggregation_type]).to eq('count_agg')
+        expect(charge_usage[:charge][:charge_model]).to eq('graduated')
+        expect(charge_usage[:units]).to eq('4.0')
+        expect(charge_usage[:amount_cents]).to eq(5)
+        expect(charge_usage[:amount_currency]).to eq('EUR')
+        expect(charge_usage[:groups]).to eq([])
+      end
+    end
+
+    context 'with one dimension group' do
+      let(:aws) { create(:group, billable_metric: metric, key: 'cloud', value: 'aws') }
+      let(:google) { create(:group, billable_metric: metric, key: 'cloud', value: 'google') }
+      let(:charge) do
+        create(
+          :standard_charge,
+          plan: subscription.plan,
+          billable_metric: metric,
+          properties: {},
+          group_properties: [
+            build(
+              :group_property,
+              group: aws,
+              values: { amount: '10', amount_currency: 'EUR' },
+            ),
+            build(
+              :group_property,
+              group: google,
+              values: { amount: '20', amount_currency: 'EUR' },
+            ),
+          ],
+        )
+      end
+
+      before do
+        create_list(
+          :event,
+          3,
+          organization:,
+          customer:,
+          subscription:,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'aws' },
+        )
+
+        create(
+          :event,
+          organization:,
+          customer:,
+          subscription:,
+          code: metric.code,
+          timestamp: Time.zone.now,
+          properties: { cloud: 'google' },
+        )
+      end
+
+      it 'returns the group usage for the customer' do
+        get_with_token(
+          organization,
+          "/api/v1/customers/#{customer.external_id}/current_usage?external_subscription_id=#{subscription.external_id}",
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)